ui/src/message_popup/popup_question.jsx

```diff
@@ -49,221 +47,90 @@ export default React.createClass({
 
   getInitialState: function() {
     return {
-      elapsedMs: 0,
-      initialResponseText: '',
-      finalResponseText: undefined,
-      isRevising: false,
-      isDoneRevising: false
+      response: null
     };
   },
-
-  componentDidMount() {
-    this.setInterval(this.updateTimer, ONE_SECOND);
-  },
-
-  updateTimer() {
-    if(!this.state.isDoneRevising){
-      this.setState({ elapsedMs: this.state.elapsedMs + ONE_SECOND });
-    }
-  },
-
-  logResponse() {
-    this.logData('message_popup_response');
-  },
-  
-  logRevision(finalText) {
-    this.logData('message_popup_revision', {finalResponseText: finalText});
-  },
-  
-  logRevisionDeclined(){
-    this.logData('message_popup_revision_declined');
-  },
   
+  // Allow pieces of the UI to log data for particular event types, along
+  // with an opaque set of params.
   logData(type, params = {}) {
-    const {elapsedMs, initialResponseText} = this.state;
     const {question, scaffolding} = this.props;
-    const {finalResponseText} = params;
     const response:Response = {
       question,
       shouldShowStudentCard: scaffolding.shouldShowStudentCard,
       helpType: scaffolding.helpType,
-      elapsedMs,
-      initialResponseText,
-      finalResponseText
+      ...params
     };
     this.props.onLog(type, response);
   },
   
-  onTextChanged({target:{value}}:TextChangeEvent) {
-    this.setState({ initialResponseText: value });
-  },
-  
-  onDonePressed(text=undefined) {
-    if(!this.props.scaffolding.shouldShowSummary){
-      this.props.onDone(this.state.elapsedMs);
-      return;
-    }
-    
-    if(this.state.isDoneRevising){
-      this.props.onDone(this.state.elapsedMs);
-      return;
-    }
-    
-    if(!this.state.isDoneRevising){
-      var finalText = (text === undefined ? this.state.initialResponseText : text);
-      this.setState({finalResponseText: finalText});
-      this.setState({isDoneRevising: true});
-      return;
-    }
-  },
-
-  onSavePressed() {
-    this.logResponse();
-    const {scaffolding} = this.props;
-    const {examples} = this.props.question;
-    if(scaffolding.helpType === 'feedback' && examples.length > 0) {
-      this.setState({ isRevising: true });
-    } else{
-      this.onDonePressed();
-    }
-  },
-  
-  onRevised(text) {
-    this.logRevision(text);
-    this.onDonePressed(text);
-  },
-  onPassed() {
-    this.logRevisionDeclined();
-    this.onDonePressed();
+  onResponseSubmitted(response) {
+    this.setState({response}, () => {
+      if (!this.props.scaffolding.shouldShowSummary) {
+        this.onDone();
+      }
+    });
   },
 
-  onRequestClose() {
-    //This empty function is meant to cancel out the closing of the toast without having to use
-    //an extremely large autoHideDuration
+  onDone() {
+    this.props.onDone(this.state.response.elapsedMs / 1000);
```

Comment:
I think `this.props.onDone` takes in milliseconds so just `this.state.response.elapsedMs` would work.